### DISCUSSION TITLE
Mark dead ROS node as stale in diagnostics panel

### DISCUSSION THREAD

**Discussion by dbruggner** [foxglove/community#750] [Q&A]
I use [ROS2 diagnostics](https://github.com/ros/diagnostics/blob/ros2/README.md) with the `diagnostic_updater` it offers a [heartbeat](https://github.com/ros/diagnostics/blob/9f402787ea2c9b3dd4d7e51a9986810e8a3400ba/diagnostic_updater/include/diagnostic_updater/update_functions.hpp#L417) `DiagnosticTask`. This task publishes 'alive' messages on `/diagnostics` as long as the ROS node runs. When the ROS node is killed Foxglove Studio still reports it as alive in the diagnostics panel. In the RQT runtime monitor these nodes are marked as stalled: https://docs.fetchrobotics.com/_images/rqt_runtime_monitor_runstopped.png

Is it possible to also add a feature to mark killed ROS nodes in the diagnostics panel?

**Comment by defunctzombie** [comment]
Anything is possible with enough code :D

It would be useful to see when messages go stale, maybe configure that timeout, or generally see when data has stopped being received when trying to debug a robot. It would be a good addition and we've noted the product feedback. Until we get around to it, we'd welcome a contribution for this if you want to dive into the code.
